### DISCUSSION TITLE
Renovate does not use GOPROXY from customEnvVariables

### DISCUSSION THREAD

**Discussion by uhthomas** [renovatebot/renovate#27002] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.156.5

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

The documentation suggests that [customEnvVariables](https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables) can be used to configure the Go environment.

See [this example](https://github.com/renovatebot/renovate/blob/6bf090642311a59986947edfeeee97effd8f391b/docs/usage/getting-started/private-packages.md?plain=1#L136-L140).

It seems that the docs are a bit inconsistent? When should customEnvVariables be used, and why doesn't it work for GOPROXY? Does it work for the other `GO*` environment variables?

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
They are used for child process only. If you control the environment then you can just set them in regular env?

**Reply by uhthomas** [reply]
Yeah, and we have done that. I guess I feel this is just confusing and whether these are set in a child process or not seems like an implementation detail without clear explanation in the docs. Does it need to be set for both? When is it be relevant for child processes and when isn't it?

Does what I'm saying make sense?

**Reply by rarkins** [reply]
My position is: I think it's unnecessary for you to put GOPROXY in customEnvVariables, but it's also confusing why it doesn't work. So please "Suggest an idea" in a new discussion which is that customEnvVariables get used in as many process.env.* places as possible in source code for consistency. It will take some time to track them all down and work out if they're applicable

**Reply by uhthomas** [reply]
I'm glad we're aligned, thank you.

I'd like to expand on that and add that it also seems confusing as to what does and what doesn't run as a child process? I suppose it doesn't matter if we're going to change customEnvVariables to be used in as many places as possible.

**Reply by rarkins** [reply]
Child process is whenever we call a third party tool like npm or poetry
